ipamapi/contract.go

```diff
@@ -42,59 +51,22 @@ type AddressSpace string
 // Common key is a addressspace
 type Config interface {
```

Comment:
I don't think driver/apps are prevented from using the config api. libnetwork would provide the driver with reference to both config and allocation providers.